src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4121,6 +4121,12 @@ public bool SendToAsync(SocketAsyncEventArgs e)
 
             // Prepare for and make the native call.
             e.StartOperationCommon(this, SocketAsyncOperation.SendTo);
+
+            if (_rightEndPoint == null)
+            {
+                _rightEndPoint = endPointSnapshot;
+            }
```

Comment:
I was not sure at which point should this happen. `BeginSendTo` assigns it before doing the actual WSA P/Invoke, I tried to follow that approach.

Reply:
Nit: for future reference, this can now be written as:
```C#
_rightEndPoint ??= endPointSnapshot;
```